pkg/registry/core/shoot/strategy_test.go

```diff
@@ -455,6 +468,66 @@ var _ = Describe("Strategy", func() {
 				shootregistry.Strategy.PrepareForUpdate(context.TODO(), newShoot, oldShoot)
 				Expect(newShoot.Generation).To(Equal(oldShoot.Generation))
 			})
+
+			It("should increase when the rotate-serviceaccount-key-start annotation gets set and feature gate is enabled", func() {
+				defer test.WithFeatureGate(utilfeature.DefaultFeatureGate, features.ShootSARotation, true)()
```

Comment:
Nit: does it make sense to use a table test? It looks like the only change between the ITs is the feature gate value and the operation annotation value on the new Shoot.